resolwe_bio/processes/expression/slamdunk.py

```diff
@@ -70,37 +80,64 @@ def run(self, inputs, outputs):
         name = basename[:-9]
 
         reads_paths = [reads.path for reads in inputs.reads.fastq] + [reads.path for reads in inputs.reads.fastq2]
-        (Cmd['ln']['-s', inputs.transcriptome.fasta.path, 'transcriptome.fasta'])()
+        (Cmd['ln']['-s', inputs.ref_seq.fasta.path, 'ref_seq.fasta'])()
         concatenated_name = f'{name}_concatenated.fastq.gz'
         (Cmd['cat'][reads_paths] > concatenated_name)()
 
         args = [
-            '-r', 'transcriptome.fasta',
+            '-r', 'ref_seq.fasta',
             '-b', inputs.regions.bed.path,
             '-o', 'slamdunk_all',
             '-n', inputs.max_alignments,
-            '-t', '8',
-            '-rl', inputs.read_length
+            '-t', self.requirements.resources.cores,
+            '-rl', inputs.read_length,
         ]
 
         if inputs.filter_multimappers:
             args.extend(['-m', '-fb', inputs.regions.bed.path])
 
         args.append(concatenated_name)
 
-        (Cmd['slamdunk']['all'][args])()
+        return_code, _, _ = Cmd['slamdunk']['all'][args] & TEE(retcode=None)
+        if return_code:
+            self.error('Slamdunk analysis failed.')
+
+        bam_path = os.path.join('slamdunk_all', 'filter', '*.bam')
+        bam_file = glob.glob(bam_path)[0]
+        if os.path.isfile(bam_file):
+            bam_name = os.path.basename(bam_file)[:-4]
+            bam_stats = bam_name + '_stats.txt'
+            (Cmd['samtools']['flagstat', bam_file] > bam_stats)()
+            Cmd['bamtobigwig.sh']([bam_file, inputs.regions.species, self.requirements.resources.cores])
+            outputs.bam = bam_file
+            outputs.bai = bam_file + '.bai'
+            outputs.stats = bam_stats
+            outputs.bigwig = bam_name + '.bw'
+        else:
+            self.error('Slamdunk failed to create a BAM file. Exiting.')
+
+        snp_path = os.path.join('slamdunk_all', 'snp', '*_snp.vcf')
```

Comment:
Viewing VCF files in genome browsers is better if the file is compressed using bgzip and indexed. If it does not brake the downstream process which requires the vcf as an input, compress and index the vcf in [this way](https://github.com/genialis/resolwe-bio/blob/master/resolwe_bio/processes/import_data/variants_vcf.yml#L71). For the index file, use the separate output field, similar as bam/bai combination

Reply:
It seems that snpeval is not able to handle compressed files properly. Despite using the same input the results differ and one of the variants is reported incorrectly.